### ISSUE TITLE
Can't Get Token after Upgrade to 2.0.4 Rest

### ISSUE THREAD

**Issue by ghost** [magento/magento2#4016]
## Steps to reproduce
1. Install Magento from release.

`composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition /var/www/domain/magento/`
1. from terminal enter ...

`curl -X POST "http://domain.com/magento/index.php/rest/V1/integration/admin/token" \`
`-H "Content-Type:application/json" -d '{"username":"admin", "password":"admin123"}'`
## Expected result

`"6kv3mxw10oxrcaqkkq956t23xhybk90q"` or similar
## Actual result
1. Magento 2.0.2 ... result as expected
2. Magento 2.0.4 ... (after either upgrade of 2.0.2 or new install)

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL /magento/index.php/rest/V1/integration/admin/token was not found on this server.</p>
<p>Additionally, a 404 Not Found
error was encountered while trying to use an ErrorDocument to handle the request.</p>
</body></html>
```

**Comment by mazhalai** [comment]
@ircsome after composer create-project, can you verify that you installed Magento2?

**Comment by ghost** [comment]
Yes ... 

@mazhalai here are the rest of the installation steps I took after composer create-project ....

```
cd /var/www/domain/magento
sudo chown -R domain:www-data .
sudo find . -type d -exec chmod 770 {} \; && find . -type f -exec chmod 660 {} \; && chmod u+x bin/magento

# made a database in phpMyAdmin 

export PATH=$PATH:/var/www/domain/magento/bin
echo '# set PATH so it includes magento bin at login
if [ -d "/var/www/domain/magento/bin" ] ; then
 PATH="$PATH:/var/www/domain/magento/bin"
fi' | sudo tee -a ~/.profile

# install magento
magento setup:install --base-url=http://domain.com/magento/ \
--db-host=localhost --db-name=magento --db-user=root --db-password=domain \
--admin-firstname=Neil --admin-lastname=Edmondson --admin-email=neil@domain.com \
--admin-user=admin --admin-password=admin123 --language=en_US \
--currency=USD --timezone=America/Chicago --cleanup-database --session-save=files --use-rewrites=1

# resulted in
# [SUCCESS]: Magento installation complete.
# [SUCCESS]: Magento Admin URI: /admin_pc6055
```

**Comment by mazhalai** [comment]
@ircsome so to recap, you deployed via `composer create-project`, installed Magneto2, and then attempted to curl details from an integration? Did you follow steps to [create and configure an integration](http://devdocs.magento.com/guides/v2.0/howdoi/webapi/integration.html)?

**Comment by ghost** [comment]
Did you follow steps to create and configure an integration?

@mazhalai - yes.  I am using Token Based Authentication as suggested in [these instructions](http://devdocs.magento.com/guides/v2.0/get-started/authentication/gs-authentication-token.html).

I am developing a php module to access product data via the rest api (see attached for a couple of functions from this).  

My primary use of curl from the terminal is to test if the api interface is working before I code the php.

I have tried to a) update the existing 2.0.2 install to 2.0.4 and re-test, b) install a completely new 2.0.4 and test.  

My testing works fine with 2.0.2 - and I get the results returned that I expect.  But I get the errors described in the original report with either install of 2.0.4.

[restapi.txt](https://github.com/magento/magento2/files/206686/restapi.txt)

**Comment by mazhalai** [comment]
@ircsome we have created MAGETWO-51785 to investigate and fix.

**Comment by ghost** [comment]
@mazhalai thanks!  how do I track the progress of MAGETWO-51785?

**Comment by mazhalai** [comment]
@ircsome  Sorry that is our internal tracking, it is not available for the public. Once a solution is merged, the commits will be linked to this ticket.

**Comment by arkadiych** [comment]
@ircsome, could you answer some questions:
- does it work for you if "--base-url=http://domain.com/" for 2.0.2 and 2.0.4
- when you get 404 via REST request, could you open home and admin pages?

**Comment by ghost** [comment]
@arkadiych
- does it work for you if "--base-url=http://domain.com/" for 2.0.2? YES
- does it work for you if "--base-url=http://domain.com/" for 2.0.4? NO
- when you get 404 via REST request, could you open home and admin pages? YES

see attached images for illustration ...
<img width="570" alt="gettoken" src="https://cloud.githubusercontent.com/assets/4860281/14785448/6cd0385a-0ac0-11e6-850a-e2e6364e55bd.PNG">
![adminpage-stotty](https://cloud.githubusercontent.com/assets/4860281/14785685/aa445724-0ac1-11e6-90af-3f2119c0e300.PNG)
![adminpage](https://cloud.githubusercontent.com/assets/4860281/14785447/6ccb4746-0ac0-11e6-901c-69b7f957a2e0.PNG)

**Comment by arkadiych** [comment]
@ircsome, if you update a magento instance on clarts.com to 2.0.4, does it work or not?
I have installed magento 2.0.4 from composer and was able to get the expected result. 
Could you also check web server settings? 
Any additional information, that can help to reproduce the issue, would be also appreciated.

**Comment by ghost** [comment]
@arkadiych

When I first reported the error I was getting 404s whether I tried against a new install of 2.0.4 or an upgrade of 2.0.2 to 2.0.4.

I have just done a clean install of 2.0.4 using the attached script.  I no longer get a 404 ... but now I don't get anything returned.  Has the repo changed in he intervening period?  (NOTE: just noticed the repo has been updated to 2.0.5 which may account for the different result).

[token.txt](https://github.com/magento/magento2/files/239700/token.txt)

**Comment by arkadiych** [comment]
@ircsome , I've installed Magento 2.0.5 from composer. I verified for php 5.6 and php 7. In both cases  it works as expected. 
I'd recommend you to check permissions. Try also to debug the code.

**Comment by mazhalai** [comment]
@ircsome we are closing this bug, since it is not reproducible after repeated tries. Please feel free to open a new issue if problems exist.
